packages/sanity/src/core/studio/packageVersionStatus/PackageVersionStatusProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import {Box, Button, Stack, useToast} from '@sanity/ui'
+import {type ReactNode, useEffect, useState} from 'react'
+import {SANITY_VERSION} from 'sanity'
+import semver from 'semver'
+
+import {hasSanityPackageInImportMap} from '../../environment/hasSanityPackageInImportMap'
+import {useTranslation} from '../../i18n'
+import {checkForLatestVersions} from './checkForLatestVersions'
+
+/*
+ * We are currently only checking to see if the sanity module has a new version available.
+ * We can add more packages to this list (e.g., @sanity/vision) if we want to check for more.
+ */
+const currentPackageVersions: Record<string, string> = {
+  sanity: SANITY_VERSION,
+}
+
+export function PackageVersionStatusProvider({children}: {children: ReactNode}) {
+  const [hasNewVersion, setHasNewVersion] = useState<boolean>(false)
+  const toast = useToast()
+  const {t} = useTranslation()
+
+  const autoUpdatingPackages = hasSanityPackageInImportMap()
+
+  const onClick = () => {
+    window.location.reload()
+  }
+
+  useEffect(() => {
+    const sub = checkForLatestVersions(currentPackageVersions).subscribe({
```

Comment:
❓would it make sense to check the version only if it's autoUpdating studios? Doesn't make sense to call the service if it's not, seems unnecessary

Reply:
ugh yes, victim of a refactor trying to get an e2e test going. sorted now.